numpy/lib/function_base.py

```diff
@@ -2536,8 +2536,11 @@ class vectorize(object):
 
     Parameters
     ----------
-    pyfunc : callable
+    pyfunc : callable, optional
         A python function or method.
+        This is optional since numpy version 1.14. When omitted, a
+        partially-initialized decorator is returned to support keyword
+        arguments when using decorator syntax.
```

Comment:
(I know there's versionadded for new arguments, but I'm not sure whether there's something similar for when a type has changed/become optional, so I wrote it in prose)